### ISSUE TITLE
No next page button?

### ISSUE THREAD

**Issue by cw-kid** [winsomehax/plugin.video.bitchute#20]
Hi

I've just installed this add-on, version 2.1.0 from the Testing branch.

I can access my subscribed channels OK, but all of them only show one page worth of videos in Kodi.

There is no next page button to list the next page of videos?

Running LibreElec Kodi 18.9 Leia.

Thanks

**Comment by winsomehax** [comment]
Yeah, I have to scrape the HTML because there's no API... and the rest of the videos aren't sent unless the user scrolls down. I didn't bother at first because I was getting it working and I was only really interested in recent videos.

It's on my list to fix

**Comment by cw-kid** [comment]
> the rest of the videos aren't sent unless the user scrolls down

If I am on the very last video in the list and press down once more, it then selects the very first video at the top of the list. 

It doesn't appear to load more videos. 

Thanks.

**Comment by winsomehax** [comment]
I know... I mean the bitchute website I am scraping from to make the menu
you see in Kodi. It doesn't load the videos until the user has moved down.

I've got to make the addon request more to see the whole list of videos. It
is on my todo list.

**Comment by cw-kid** [comment]
Got you, I understand what you meant now.

Yes more videos returned in to the lists in Kodi would be great. 

Thank you.

**Comment by winsomehax** [comment]
Give the Testing branch a try please, if you don't mind. I've added a version of previous/next. So see how that works.

**Comment by cw-kid** [comment]
Seems to be working well, it loads more pages of videos when selecting "Next Page". 

Thanks

**Comment by cw-kid** [comment]
Hi 

Any news on adding the Next Page feature to the other areas of the plugin ? 

Thanks

**Comment by Slavegrid** [comment]
Had Deepseek Flash add pagination where possible with Commit 9a84921
